.github/workflows/nightly_release.yml

```
@@ -0,0 +1,100 @@
+name: SalesforceCommerceCloud/pwa-kit/nightly_release
+on:
+  workflow_dispatch:
+  schedule:
+    # Run weekdays at 12am (PST) - cron uses UTC times
+    - cron: "0 8 * * 1-5"
+
+jobs:
+  create_nightly_release:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Get Timestamp
+        run: |-
+          echo "release_timestamp=$(date +'%Y%m%d%H%M%S')" >> "$GITHUB_ENV"
+
+      - name: Get monorepo version
+        run: |-
+          version=`jq -r ".version" package.json | cut -d "-" -f 1`
+          echo "monorepo_version_base=$version" >> "$GITHUB_ENV"
+          version_major=`echo "$version" | cut -d "." -f 1`
+          echo "monorepo_version_major=$version_major" >> "$GITHUB_ENV"
+
+      - name: Get retail-react-app version
+        run: |-
+          version=`jq -r ".version" packages/template-retail-react-app/package.json | cut -d "-" -f 1`
+          echo "retail_app_version_base=$version" >> "$GITHUB_ENV"
+
+      - name: Get commerce-sdk-react version
+        run: |-
+          version=`jq -r ".version" packages/commerce-sdk-react/package.json | cut -d "-" -f 1`
+          echo "commerce_sdk_react_version_base=$version" >> "$GITHUB_ENV"
+
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+          cache: npm
+
+      - name: Install Monorepo Dependencies
+        run: |-
+          # Install node dependencies
+          node ./scripts/gtime.js monorepo_install npm ci
+
+      - name: Create Release Branch
+        run: git switch -c release-nightly-${{ env.monorepo_version_major }}.x
```

```suggestion
        run: git switch release-nightly-${{ env.monorepo_version_major }}.x
```
❗ `-c` will fail if the branch already exists! Releasing a new major is something we know of well in advance, so we can easily create a new branch for it ahead of time.

---

```
git switch develop
git merge -s ours release-nightly-${{ env.monorepo_version_major }}.x
git switch release-nightly-${{ env.monorepo_version_major }}.x
git merge develop
```
This will merge `develop` into the nightly branch, preferring the changes from `develop` for all merge conflicts.

(It would be nice if we could do `git merge -s theirs develop`, but that doesn't exist, so we just have to flip it around, briefly.)

---

Updated to use fallback for git switch to create a new branch if one does not exisst. Also updated the merge commands to pull changes from develop.